### ISSUE TITLE
window.open blocked in Facebook/Messenger in-app browser

### ISSUE THREAD

**Issue by subpublic** [sahat/satellizer#810]
We are posting our registration page on facebook and messenger apps, but the loginbutton doesn't work  on iOS. It works fine if opened in Safari.

My guess is that the in-app browser is blocking popups.

Anyone else with this problem?
Solutions? I'm thinking maybe using a fullscreen iframe instead of window.open.

I'm using version 0.13.1

**Comment by subpublic** [comment]
The in-app browser actually opens the page (doesn't block it), but not in another window. It just goes forward so the polling stops isn't possible to bli closed by satellizer.

**Comment by garrettlangley** [comment]
@subpublic what did you wind up doing here?

**Comment by subpublic** [comment]
We wounded up detecting iphone FB-apps and saving our login details in localStorage.
And when returning to the redirectUrl we check the localStorage if anything is saved, here calling the same registration as we do after the regular satellizer login.

This detects both FB and Messenger app in iphone (android doesn't have the problem)
```
if (navigator.userAgent.match(/iPhone/i) && navigator.userAgent.match(/FBMD/i)) {
```

**Comment by garrettlangley** [comment]
but @subpublic how did you avoid the the popup? did you do an iframe?

**Comment by subpublic** [comment]
No need to, it doesn't open any popup, it just opens the loginprovider (fb/google/etc) in the same window.
Kind of the same solution as if we didn't use satellizer at all.

**Comment by garrettlangley** [comment]
so did you fork satellizer and make changes? or just drop satellizer entirely?

**Comment by subpublic** [comment]
No, it's no real solution in satellizer. Just needs do be handled separately.
Still using satellizer when not in iphone/fb.

**Comment by kirill-sokolov** [comment]
@subpublic hi, men! i have such problem as you - i do login via satelizzer, it all works great on desktops, but on ipad i have trubbles - facebook authorization popup opens, i do login, and then it yust closes and satelizer does not do anything.
i mean -
 function authenticate(provider) {
       return $auth.authenticate(provider);
  }

**Comment by bmarotta** [comment]
@subpublic : Could you please publish somewhere a full code snippet of your workaround?
